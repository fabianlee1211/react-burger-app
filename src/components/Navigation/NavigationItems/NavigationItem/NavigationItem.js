import React from 'react';
import { Link } from 'react-router-dom';
import classes from './NavigationItem.css';

const NavigationItem = (props) => (
  <li className={classes.NavigationItem}>
    <Link to={props.link} className={props.active ? classes.active : null}>{props.children}</Link>
  </li>
);

export default NavigationItem;