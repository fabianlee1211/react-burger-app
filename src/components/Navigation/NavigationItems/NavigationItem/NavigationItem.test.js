import React from 'react';
import { shallow } from 'enzyme';
import NavigationItem from './NavigationItem';

describe('<NavigationItem />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<NavigationItem link='/' />);
  });

  it('renders NavigationItem correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has NavigationItem class', () => {
    expect(wrapper.find('.NavigationItem')).toHaveLength(1);
  });

  it('renders NavigationItem correctly with given props', () => {
    wrapper.setProps({ link: '/home', exact: false, children: 'Hello' });
    expect(wrapper.find('NavLink').props().to).toBeDefined();
    expect(wrapper.find('NavLink').props().exact).toBeFalsy();
    expect(
      wrapper
        .find('NavLink')
        .children()
        .text()
    ).toEqual('Hello');
  });
});
