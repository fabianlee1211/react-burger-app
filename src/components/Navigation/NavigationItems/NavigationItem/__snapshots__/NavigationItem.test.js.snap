// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItem /> renders NavigationItem correctly 1`] = `
<li
  className="NavigationItem"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/"
  />
</li>
`;
